### ISSUE TITLE
Submission checklist for GRNsight manuscript

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#267]
- [x] upload .docx version of manuscript to repository
- [x] @dondi, answer highlighted blue questions in manuscript (including blue references)
- [x] @kdahlquist, redo Figure 1, upload to repository
- [x] @kdahlquist, redo Figure 2, upload to repository
- [x] @kdahlquist, redo Figure 5, upload to repository
- [x] check reference formatting, find DOI's (all references added to @kdahlquist's Zotaro, exported in PeerJ format)
- [x] review abstract and edit, if needed once main text is complete

**Comment by kdahlquist** [comment]
I uploaded the manuscript.  Please make any revisions in "track changes" mode in Word.  @dondi , the stuff highlighted in blue has questions/comments, mostly for you.  The yellow highlighting is for references.

Figures 1 and 2 have been uploaded in several versions.  "zoom100" or "zoom145" refers to the zoom level in Excel used when taking the screenshot.  I'm not sure which will be better.  I have also uploaded the Excel file used to make the screenshots, the .psd, .jpg, and .png (preferred by PeerJ) versions of the files in case we need them.

I changed the filenames for the architecture and interface screenshots to include the figure numbers as I don't think we are going to change the order of the figures.

**Comment by kdahlquist** [comment]
I have been working on Figure 5.  I've committed the individual parts to the repository; I'm still fiddling with the layout of figure 5E and to some extent figure 5B, but I need to go today.  Once I've finalized it, I will compile the final figure and put the letter labels on it for the parts.

**Comment by kdahlquist** [comment]
Figure 5 is done and pushed to the repository.  I need to delete some of the individual part files that were not used, but it's done.  Moving on to formatting references.

**Comment by kdahlquist** [comment]
I'm pretty much done with my tasks.  Waiting on @dondi and @bengfitzpatrick to review, give feedback, and edit, if necessary.  All updated files have been pushed to the repository here:

https://github.com/dondi/GRNsight/tree/peerj/documents/PeerJ

**Comment by dondi** [comment]
I’ve done an initial pass of the paper; many of the blue highlights were addressed. Minor edits happened here and there. One loose end: for the testing results which were noted as coming from beta, the intent is to replace these completely with the current master, yes? Also, for the note re: why D3.js, I added some sentences that may need additional review. (track changes is active)

**Comment by kdahlquist** [comment]
Going through your changes now and merging with some that @bengfitzpatrick sent me via e-mail.  Quick question.  I'm assuming that if you didn't change it (blue text), you're OK with it.  I'll comment when I've committed new changes.

**Comment by dondi** [comment]
Yes, if blue text is untouched, that means the text can stay. I think there are 1 or 2 where I attached a comment or question, but otherwise untouched blue was OK with me. I figured the blue would be easy to spot in case you wanted to review any nearby text that was edited. I also changed blue citations to yellow once I finalized the entry in the bibliography.

**Comment by kdahlquist** [comment]
OK, so I've pushed the manuscript and references file to the repository, incorporating both @dondi and @bengfitzpatrick 's comments.  A few things to note:

1. @bengfitzpatrick 's comments were on a much older version of the manuscript, a lot of the changes he suggested were superseded by changes already made.  In a lot of the cases, they were similar to changes he suggested, so great minds think alike!
2. I'm managing references using Zotaro, so I redid the ones that @dondi added based on the bibliographic tool.  I'm recording the ISBN's for books, even though it is not specifically requested by PeerJ, so there is no ambiguity (like using the DOI's).
3. What's left in terms of comments/formatting:
 - [ ] The blue text in the abstract was to highlight the sentence I added for your approval, @dondi.  If you are OK with it, just undo the highlighting and delete the comment.
 - [ ] Yes, I think we should change the numbers for tests/files based on master.  We could add a sentence that testing and development is ongoing and that beta has more, but I think it will be more clear if we just refer to the master branch.
 - [ ] Agenda for tomorrow's face-to-face meeting is to discuss any last-minute changes to the text; I think it's ready for submission.  I have approval from all the co-authors to submit and I want to get that taken care of this week.
 - [ ] There is a reference in the text to the YEASTRACT instructions appearing on the GRNsight documentation page; we need to actually move this over, it is issue #275 .
 - [ ] Let's review the other open related issues, persistant URL #266 , privacy policy, #258, and any others we think we need to take care of right away.
 - [ ] If there's time, we can talk about summer plans, post-manuscript submission.

**Comment by kdahlquist** [comment]
Ack.  My revised docx is not showing up in the repository, even though the push seemed to work.  I'm going to try again, but I don't really know what happened.

**Comment by kdahlquist** [comment]
OK, now the new version is really up there.  Of course, I fixed this by recloning the repository...

**Comment by dondi** [comment]
Blue highlights all handled with my latest push, including test coverage numbers now based on master.

**Comment by kdahlquist** [comment]
I have accepted @dondi 's changes and pushed to repository.

**Comment by kdahlquist** [comment]
I'm going through the submission process on the PeerJ web site right now.  Some items for potential discussion:
1. Subject areas: must have at least 2, I chose "bioinformatics", "graphics", and "software engineering" from the list shown here:  https://peerj.com/subjects/cs  We are allowed up to 5.  Do you want to add anything?
2. Keywords: we are allowed up to 10 keywords that we make up ourselves.  So far I've come up with
* gene regulatory networks
* visualization
* web application
* web service
* automatic graph layout
Do you want to add anything?
3. We need to suggest at least 2 reviewers.  I will look into this today, any thoughts?
* best practices for scientific computing

**Comment by kdahlquist** [comment]
On the declarations page of the submission, it asks the following questions:
1. Which authors conceived and designed the experiments?
2. Which authors performed the experiments?
3. Which authors analyzed the data?
4. Which authors contributed reagents/materials/analysis tools?
5. Which authors wrote the manuscript?
6. Which authors prepared the figures and/or tables?
7. Which authors wrote the code, designed the software or performed the computation work?
8. Which authors reviewed drafts of the paper?
9. Which authors made other contributions?

There are then checkboxes with all the author names or "none", so I guess we can just choose "none" if we feel the category doesn't apply.  I don't think 4 or 9 apply to us, but I'm torn as to whether 1, 2, or 3 apply or that because this is a software paper it all gets folded into 7.

**Comment by dondi** [comment]
Regarding subject areas, I looked at the list and I didn’t see anything further to add. I would have wanted “information visualization” but they don’t have that (“visual analytics” is related but not applicable to us directly).

Nothing further for the keywords; I think the current list captures it.

No specific reviewers in mind either. I figure these would be more on the bioinformatics side rather than pure computer science, and no one specific comes to mind that you might not already mention (John Jungck, Mark LeBlanc?).

Finally with regard to the declarations, strange that it is all or none. Yes 4 and 9 probably don't apply, unless GRNmap is considered to be an "analysis tool" in which case then 4 would count. I think 1, 2, and 3 can be interpreted as considering GRNs with expected properties to be "experiments," in which case we did specify certain GRNs that we wanted to see, then looked at the resulting visualizations to see if they met our needs or displayed correctly. Those are distinct from the actual software development, I’d say?

**Comment by kdahlquist** [comment]
Sorry, I didn't mean to apply that the declarations are all or none, you can check specific boxes for specific authors.  I just meant that items 1, 2, and 3 seem to apply more to a biology paper than to us.  I can read them either way, but I'm leaning to including them as you suggest, i.e., not clicking "none" for them.

**Comment by dondi** [comment]
Ah I see. But yes, I do still agree that 1, 2, and 3 can be considered as “not none.” (I can also go either way, and like you said it’s more of leaning toward it)

**Comment by kdahlquist** [comment]
I also forgot to ask you about Figures 1 and 2.  There are two different versions for each that were made at 100% zoom and 145% zoom in Excel before making the screenshot.  They are saved as .png, .jpg, and .psd in the repository, although the journal is only going to want them as .png.  Can you take a look and recommend which one to use?

**Comment by dondi** [comment]
The 145s look better to my eyes (I only looked at the .pngs since that is what they will take).

**Comment by kdahlquist** [comment]
OK, 145s it is.  

They are counting characters instead of words for their abstract count and we are over by 191 characters.  I'm going to work on paring it down now.

**Comment by kdahlquist** [comment]
Something else we should know about... http://biojs.net

**Comment by kdahlquist** [comment]
Manuscript submitted!  I'm closing this and will open a new issue to follow our progress through the review process.  Congratulations to us!
